### PR TITLE
Document general error codes

### PR DESCRIPTION
**Pull request by DoodleBobBuffPants**

### CODE REVIEW

File: documentation/usage/errorcodes/index.md [hunk 1]
```diff
@@ -16,6 +16,38 @@ These error codes may appear from any command, and are indicative of more genera
 - **Caused by:** An unexpected error within Flyway (e.g. a null pointer exception)
 - **Solution:** Please contact support or create a [GitHub issue](https://github.com/flyway/flyway/issues)
 
+### `ERROR`
+- **Caused by:** An unexpected error within Flyway (e.g. a null pointer exception)
```
Comment (MikielAgutu) [inline documentation/usage/errorcodes/index.md]: When we first designed this feature, we had in mind the distinction between an error and a fault.

A fault is an unexpected failure - like a bug. The software has gone wrong even with valid input. On the other hand, an error is a failure state due to invalid user input (e.g. providing invalid schema name).

We used this as inspiration: https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#5-taxonomy

(Most of this context might be lost / buried in github issues, but this is what I remember).

Now perhaps the code itself doesn't abide by this distinction, but it ought to.

**Review by MikielAgutu — CHANGES_REQUESTED** [review]

File: documentation/usage/errorcodes/index.md (hunk 1 above)
Comment (MikielAgutu) [reply documentation/usage/errorcodes/index.md]: Ah yes, here is the issue: https://github.com/red-gate/flyway-main/pull/53#issue-513350304

> My idea is to release something simple and gather feedback from the Orcas. We can then adjust this feature based on the feedback.

Well it took ~over a year~ almost 2 years, and a different team. But we got feedback eventually 😅

Comment (DoodleBobBuffPants) [reply documentation/usage/errorcodes/index.md]: see latest commit

File: documentation/usage/errorcodes/index.md [hunk 2]
```diff
@@ -16,6 +16,38 @@ These error codes may appear from any command, and are indicative of more genera
 - **Caused by:** An unexpected error within Flyway (e.g. a null pointer exception)
 - **Solution:** Please contact support or create a [GitHub issue](https://github.com/flyway/flyway/issues)
 
+### `ERROR`
+- **Caused by:** An error due to invalid configuration or usage
```
Comment (MikielAgutu) [inline documentation/usage/errorcodes/index.md]: I think we should call out that the `Error` code is a catch all, for when we don't have a more specific code to produce. So if users are switching on error types, this is essentially the default case.

Comment (DoodleBobBuffPants) [reply documentation/usage/errorcodes/index.md]: see latest commit

**Review by MikielAgutu — APPROVED** [review]

**Pull request merged**